### PR TITLE
Allow Scratch run time

### PR DESCRIPTION
**Pull request by leduythuccs**
Use our [Scratch run](https://github.com/VNOI-Admin/scratch-run) to run the scratch project.

Since scratch's project is not a text file, we pass a meta to it, if the meta has the `file-only` attribute, we assume that the source code is a URL, and we use that URL to get the real submission.

### CODE REVIEW

File: .docker/tier1/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,17 @@
 FROM dmoj/runtimes-tier1
 
 ARG TAG=master
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends unzip && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (leduythuccs) [inline .docker/tier1/Dockerfile:7]: wut is this @hieplpvip

File: .docker/Makefile [hunk 2]
```diff
@@ -6,10 +6,10 @@ TAG ?= latest
 all: judge-tier1 judge-tier2 judge-tier3
 
 judge-tier1:
-	cd tier1 && docker build --build-arg TAG="${GIT_TAG}" -t dmoj/judge-tier1 -t dmoj/judge-tier1:$(TAG) .
+	cd tier1 && docker build --build-arg TAG="${GIT_TAG}" -t vnoj/judge-tier1 -t vnoj/judge-tier1:$(TAG) .
```
Comment (leduythuccs) [inline .docker/Makefile:9]: xD I think we should keep this the old dmoj name

File: .docker/tier1/Dockerfile (hunk 1 above)
Comment (hieplpvip) [reply .docker/tier1/Dockerfile:7]: Don't install recommended packages and clean up to reduce image size (in theory)

File: .docker/Makefile (hunk 2 above)
Comment (hieplpvip) [reply .docker/Makefile:9]: We made several important changes to our fork so I think we should change the name to avoid any mistakes later.

Comment (leduythuccs) [reply .docker/Makefile:9]: make sense.

**Review by hieplpvip — APPROVED** [review]

**Pull request merged**